### PR TITLE
Leaves - Georgina

### PR DESCRIPTION
**Pull request by geomsb**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | It manages the logic to interact with the data.
Describe in your own words what the Controller is doing in Rails | It controls the flow between the view and the models.
Describe in your own words what the View is doing in Rails | It displays logic and turns data to html
Describe an edge-case controller test you wrote | it "can't update an existing task giving wrong parameters": if you give a wrong parameter to update a task, the task is not going to take that parameter, it would be nil. 
What is the purpose of using strong params? (i.e. the params method in the controller) | To prevent errors. Controllers are not permitted to pass parameters directly so we need to use them.
How are Rails migrations related to Rails models? | because both are related with the information that the database contains.
Describe one area of Rails that are still unclear on | tests related edge cases.

### CODE REVIEW

File: test/controllers/tasks_controller_test.rb [hunk 1]
```diff
@@ -90,37 +81,108 @@
...
+        task: {
+          names: "task_70",
+          description: "clean the house",
+          completed: nil,
+        } 
+      }
+      
+      patch task_path(existing_task.id), params: updated_task_form_data
+
+      expect(Task.find_by(id: existing_task.id).name).must_equal nil
```
Comment (dHelmgren) [inline test/controllers/tasks_controller_test.rb:134]: This throws a warning!

`.DEPRECATED: Use assert_nil if expecting nil from test/controllers/tasks_controller_test.rb:134. This will fail in Minitest 6.
.......DEPRECATED: Use assert_nil if expecting nil from test/controllers/tasks_controller_test.rb:74. This will fail in Minitest 6.`

File: test/controllers/tasks_controller_test.rb [hunk 2]
```diff
@@ -90,37 +81,108 @@
...
+        delete task_path(old_task.id)
+      }.must_change "Task.count", 1
+
+    end
   end
 
   # Complete for Wave 4
   describe "toggle_complete" do
-    # Your tests go here
+    it "can mark a task as completed" do
```
Comment (dHelmgren) [inline test/controllers/tasks_controller_test.rb:170]: What if the ID doesn't exist?

File: test/controllers/tasks_controller_test.rb [hunk 3]
```diff
@@ -90,37 +81,108 @@
...
+    it "can delete a task" do
+      # Arrange
+      task_hash = {
+        task: {
+          name: "new task",
+          description: "new task description",
+          completed: nil
+        }
+      }
+
```
Comment (dHelmgren) [inline test/controllers/tasks_controller_test.rb:157]: What if the ID doesn't exist?

File: config/routes.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  resources :tasks
+  
+  # get '/tasks', to: 'tasks#index', as: 'tasks'
+  # get '/tasks/new', to: 'tasks#new', as: 'new_task'
+  # post '/tasks', to: 'tasks#create'
+  
+  # get '/tasks/:id', to: 'tasks#show', as: 'task'
+  # get '/tasks/:id/edit', to: 'tasks#edit', as: 'edit_task'
+  # patch '/tasks/:id', to: 'tasks#update'
+  # delete '/tasks/:id', to: 'tasks#destroy'
```
Comment (dHelmgren) [inline config/routes.rb:14]: In the future, don't leave this in!

**Comment by dHelmgren** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | yes
Answered comprehension questions | yes
Successfully handles: Index, Show | yes
Index & Show tests pass | yes
Successfully handles: New, Create | yes
New & Create tests pass | yes
Successfully handles: Edit, Update | yes
Tests for Edit & Update test valid & invalid task ids | yes
Successfully handles: Destroy, Task Complete | yes
Tests for Destroy & Task Complete include tests for valid and invalid task ids | see comments
Routes follow RESTful conventions | yes
Uses named routes (like `_path`) | yes
**Overall** | Good job on Task List! It functions as expected. You hit all of the learning goals: using partials, using strong params, and following Rails conventions.
